### PR TITLE
Refactor cli serve command

### PR DESCRIPTION
**Pull request by Tobbe**
The main serve file was getting unwieldy. And the serveHandler file was pretty hard to read too. So I reorganized the code into more files to make it easier to read the code.

There is one side effect of this I wanted to mention: Now if you run `yarn rw serve api` and have an experimental server file it'll dynamically include `serveApiHandler`, which includes the `@redwoodjs/fastify` package. Previously you didn't have to pay that import cost in this specific scenario.

People running the api server in prod though should not be using our entire cli if they're worried about file sizes/mem usage. So I think it's fine. This will also go away when we move the server file out of experimental.

### CODE REVIEW

File: packages/cli/src/commands/serveBothHandler.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+      ],
+      {
+        cwd: getPaths().base,
+        stdio: 'inherit',
+        shell: true,
+      }
+    )
+    await execa(
+      'yarn',
+      ['node', path.join('dist', 'server.js'), '--enable-web'],
```
Comment (Josh-Walker-GM) [inline packages/cli/src/commands/serveBothHandler.js]:
```suggestion
      ['node', path.join('dist', 'server.js')],
```

Do we want to pass this `--enable-web` flag to the api side for the case of both and with RSC or SSR enabled? Passing this flag will make the api server use the web port and register the web plugin - so serve the static files and setup a proxy.

File: packages/cli/src/commands/serveWebHandler.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { getPaths } from '@redwoodjs/project-config'
+
+export const webSsrServerHandler = async () => {
+  await execa('yarn', ['rw-serve-fe'], {
+    cwd: getPaths().web.base,
+    stdio: 'inherit',
+    shell: true,
+  })
+}
+
```
[the remark this answers is not in the document]
Comment (Tobbe) [reply packages/cli/src/commands/serveWebHandler.js:12]: This one handles both SSR and RSC

**Review by Josh-Walker-GM — APPROVED** [review]
I have no other questions, it looks good to me.

**Comment by jtoar** [comment]
The impetus here was mainly because it builds off streaming and RSC code so it probably doesn't belong in the next release

**Pull request merged**